src/components/input/select/index.tsx

```diff
@@ -188,6 +189,13 @@ export class SmoothlyInputSelect implements Input {
 			}
 		}
 		this.items = items
+		if (this.menuHeight?.endsWith("items") && this.items.length)
+			this.element?.style.setProperty(
+				"--menu-height",
+				`${this.items[0].clientHeight * +this.menuHeight.replace(/[a-z]/g, "")}px`
+			)
+		else if (this.menuHeight)
+			this.element?.style.setProperty("--menu-height", this.menuHeight)
```

Comment:
I believe we would prefer to only have the else if part as an if like this
```ts
if (this.menuHeight)
			this.element?.style.setProperty("--menu-height", !this.menuHeight.endsWith("items") || !this.items.length ? this.menuHeight : `${this.items[0].clientHeight * +this.menuHeight.replace(/[a-z]/g, "")}px`)
```
I  also think it would be clearer to have a regex for what were are looking for rather than not looking for
`+this.menuHeight.match(/^\d+\.?\d*/)[0]`